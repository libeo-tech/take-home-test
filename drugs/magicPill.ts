import { Drug, DrugName } from "./drug";

export class MagicPill extends Drug {
    constructor(expiresIn: number, benefit: number) {
        super(DrugName.MAGIC_PILL, expiresIn, benefit);
    }

    public updateDrug(): void {}
}