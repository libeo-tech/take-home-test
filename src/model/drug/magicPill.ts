import { DrugName } from '../../type/DrugName';
import { Drug } from './drug';

export class MagicPill extends Drug {
  constructor(expiresIn: number, benefit: number) {
    super(DrugName.MAGIC_PILL, expiresIn, benefit);
  }

  public isExpired(): boolean {
    return false;
  }

  public updateValues() {
    return this;
  }
}
