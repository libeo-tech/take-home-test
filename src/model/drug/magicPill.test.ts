import { MagicPill } from './magicPill';

describe('Magic Pill', () => {
  let magicPill: MagicPill;

  beforeEach(() => {
    magicPill = new MagicPill(30, 5);
  });

  it('should not allow for a negative benefit', () => {
    magicPill.benefit = -5;
    expect(magicPill).toMatchSnapshot();
  });

  it('should not allow for a benefit greater than 50', () => {
    magicPill.benefit = 51;
    expect(magicPill).toMatchSnapshot();
  });

  it('should return expiration status', () => {
    expect(magicPill.isExpired()).toBe(false);
    magicPill.expiresIn = 0;
    expect(magicPill.isExpired()).toBe(false);
  });

  it('should not change expiration or benefit with time', () => {
    magicPill.updateValues();
    expect(magicPill).toMatchSnapshot();

    magicPill.expiresIn = 0;
    expect(magicPill).toMatchSnapshot();

    magicPill.expiresIn = 30;
    expect(magicPill).toMatchSnapshot();
  });

  it('should be serializable to JSON', () => {
    expect(magicPill.toJSON()).toMatchSnapshot();
  });

  it('should be serializable to string', () => {
    expect(magicPill.toString()).toMatchSnapshot();
  });
});
