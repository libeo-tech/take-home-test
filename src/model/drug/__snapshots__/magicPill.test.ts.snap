// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Magic Pill should be serializable to JSON 1`] = `
Object {
  "benefit": 5,
  "expiresIn": 30,
  "name": "Magic Pill",
}
`;

exports[`Magic Pill should be serializable to string 1`] = `"{\\"name\\":\\"Magic Pill\\",\\"expiresIn\\":30,\\"benefit\\":5}"`;

exports[`Magic Pill should not allow for a benefit greater than 50 1`] = `
Object {
  "benefit": 50,
  "expiresIn": 30,
  "name": "Magic Pill",
}
`;

exports[`Magic Pill should not allow for a negative benefit 1`] = `
Object {
  "benefit": 0,
  "expiresIn": 30,
  "name": "Magic Pill",
}
`;

exports[`Magic Pill should not change expiration or benefit with time 1`] = `
Object {
  "benefit": 5,
  "expiresIn": 30,
  "name": "Magic Pill",
}
`;

exports[`Magic Pill should not change expiration or benefit with time 2`] = `
Object {
  "benefit": 5,
  "expiresIn": 0,
  "name": "Magic Pill",
}
`;

exports[`Magic Pill should not change expiration or benefit with time 3`] = `
Object {
  "benefit": 5,
  "expiresIn": 30,
  "name": "Magic Pill",
}
`;
